noir-projects/aztec-nr/aztec/src/keys/public_keys.nr

```diff
@@ -39,78 +115,49 @@ impl PublicKeys {
             if is_empty(self) {
             0
         } else {
-            poseidon2_hash_with_separator(
-                [
-                self.npk_m.x,
-                self.npk_m.y,
-                self.npk_m.is_infinite as Field,
-                self.ivpk_m.x,
-                self.ivpk_m.y,
-                self.ivpk_m.is_infinite as Field,
-                self.ovpk_m.x,
-                self.ovpk_m.y,
-                self.ovpk_m.is_infinite as Field,
-                self.tpk_m.x,
-                self.tpk_m.y,
-                self.tpk_m.is_infinite as Field
-            ],
-                GENERATOR_INDEX__PUBLIC_KEYS_HASH as Field
-            )
+            poseidon2_hash_with_separator(self.serialize(), GENERATOR_INDEX__PUBLIC_KEYS_HASH as Field)
         }
         )
     }
-
-    pub fn get_key_by_index(self, index: Field) -> Point {
-        assert(index as u8 < NUM_KEY_TYPES, "Invalid key index");
-        if index == NULLIFIER_INDEX {
-            self.npk_m
-        } else if index == INCOMING_INDEX {
-            self.ivpk_m
-        } else if index == OUTGOING_INDEX {
-            self.ovpk_m
-        } else {
-            self.tpk_m
-        }
-    }
 }
 
 impl Serialize<PUBLIC_KEYS_LENGTH> for PublicKeys {
     fn serialize(self) -> [Field; PUBLIC_KEYS_LENGTH] {
         [
-            self.npk_m.x,
-            self.npk_m.y,
-            self.npk_m.is_infinite as Field,
-            self.ivpk_m.x,
-            self.ivpk_m.y,
-            self.ivpk_m.is_infinite as Field,
-            self.ovpk_m.x,
-            self.ovpk_m.y,
-            self.ovpk_m.is_infinite as Field,
-            self.tpk_m.x,
-            self.tpk_m.y,
-            self.tpk_m.is_infinite as Field
+            self.npk_m.inner.x,
+            self.npk_m.inner.y,
+            self.npk_m.inner.is_infinite as Field,
+            self.ivpk_m.inner.x,
+            self.ivpk_m.inner.y,
+            self.ivpk_m.inner.is_infinite as Field,
+            self.ovpk_m.inner.x,
+            self.ovpk_m.inner.y,
+            self.ovpk_m.inner.is_infinite as Field,
+            self.tpk_m.inner.x,
+            self.tpk_m.inner.y,
+            self.tpk_m.inner.is_infinite as Field
         ]
```

Comment:
Could we not do this as the concatenation of the serialize fns? It sounds like with `static_assert` and fixed size arrays Noir should be able to deal with this, specially with arithmetic on numeric generics. If we don't do this then I'd consider adding a comment explaining why.

Reply:
I don't follow how that would work. I can only think of having a for-loop there and building the array with that. CAn you expand?

Will do that in a separate PR if it comes to it so that I can merge this now.